lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/REEFMessageCodec.java

```diff
@@ -52,9 +52,7 @@ private REEFMessageCodec() {
   public GeneratedMessage decode(final byte[] bytes) {
     try {
       final REEFProtocol.REEFMessage message = REEFProtocol.REEFMessage.parseFrom(bytes);
-      if (message.hasJobSubmission()) {
-        return message.getJobSubmission();
-      } else if (message.hasJobControl()) {
+      if (message.hasJobControl()) {
```

Comment:
None come to mind. Still, we should keep the change minimal here. Re-thinking the protocol layer should be done separately from these changes.